File: heart/src/main/java/com/vantagecircle/heartrate/fragment/FragmentComponent.java
package com.vantagecircle.heartrate.fragment;

import com.vantagecircle.heartrate.AppComponent;
import com.vantagecircle.heartrate.fragment.ui.HeartFragment;
import com.vantagecircle.heartrate.fragment.ui.HistoryFragment;
import com.vantagecircle.heartrate.annotation.PerFragment;

import dagger.Component;

/**
 * Created by bapidas on 14/11/17.
 */
@PerFragment
@Component(dependencies = AppComponent.class, modules = FragmentModule.class)
public interface FragmentComponent {
    void inject(HeartFragment heartFragment);
    void inject(HistoryFragment historyFragment);
}

File: app/src/main/java/com/bapidas/heartrate/di/component/FragmentComponent.kt
package com.bapidas.heartrate.di.component

import com.bapidas.heartrate.di.annotation.PerFragment
import com.bapidas.heartrate.di.module.FragmentModule
import com.bapidas.heartrate.ui.fragment.ui.HeartFragment
import com.bapidas.heartrate.ui.fragment.ui.HistoryFragment
import dagger.Component

/**
 * Created by bapidas on 14/11/17.
 */
@PerFragment
@Component(dependencies = [AppComponent::class], modules = [FragmentModule::class])
interface FragmentComponent {
    fun inject(heartFragment: HeartFragment)

    fun inject(historyFragment: HistoryFragment)
}